packages/math-extras/src/pointExtras.ts

```diff
@@ -0,0 +1,419 @@
+    },
+
+    /**
+     * Computes the square magnitude of this point.
+     * If you are comparing the lengths of vectors, you should compare the length squared instead
+     * as it is slightly more efficient to calculate.
+     *
+     * Defined as the sum of the squares of each component.
+     *
+     * _Note: Only available with **@pixi/math-extras**._
+     *
+     * @method magnitudeSquared
+     * @memberof PIXI.Point#
+     * @returns {number} The magnitude squared (length squared) of the vector.
+     */
+    /**
+     * Computes the square magnitude of this point.
+     * If you are comparing the lengths of vectors, you should compare the length squared instead
+     * as it is slightly more efficient to calculate.
+     *
+     * Defined as the sum of the squares of each component.
+     *
+     * _Note: Only available with **@pixi/math-extras**._
+     *
+     * @method magnitudeSquared
+     * @memberof PIXI.ObservablePoint#
+     * @returns {number} The magnitude squared (length squared) of the vector.
+     */
+    magnitudeSquared(): number
+    {
+        return (this.x * this.x) + (this.y * this.y);
+    },
+
+    /**
+     * Computes vector projection of `this` on `onto`.
+     *
+     * Imagine a light source, parallel to `onto`, above `this`.
+     * The light would cast rays perpendicular to `onto`.
+     * `this.project(onto)` is the shadow cast by `this` on the line defined by `onto` .
+     *
+     * _Note: Only available with **@pixi/math-extras**._
+     *
+     * @method project
+     * @memberof PIXI.Point#
+     * @param {IPointData} onto - A non zero vector describing a line on which to project `this`.
+     * @param {IPointData} [outPoint] - A Point-like object in which to store the value,
+     * optional (otherwise will create a new Point).
+     * @returns {IPointData} The `this` on `onto` projection.
+     */
+    /**
+     * Computes vector projection of `this` on `onto`.
+     *
+     * Imagine a light source, parallel to `onto`, above `this`.
+     * The light would cast rays perpendicular to `onto`.
+     * `this.project(onto)` is the shadow cast by `this` on the line defined by `onto` .
+     *
+     * _Note: Only available with **@pixi/math-extras**._
+     *
+     * @method project
+     * @memberof PIXI.ObservablePoint#
+     * @param {IPointData} onto - A non zero vector describing a line on which to project `this`.
+     * @param {IPointData} [outPoint] - A Point-like object in which to store the value,
+     * optional (otherwise will create a new Point).
+     * @returns {IPointData} The `this` on `onto` projection.
+     */
+    project<T extends IPointData>(onto: IPointData, outPoint?: T): T
+    {
+        if (!outPoint)
+        {
+            (outPoint as any) = new Point();
+        }
+        // Math says: a Projected over b = [(a·b) / (b·b)] * b;
+        const scalarProjection = ((this.x * onto.x) + (this.y * onto.y)) / ((onto.x * onto.x) + (onto.y * onto.y));
```

Comment:
This actually doesn't do the formula in the comment above. It is actually `(a*b) / (b*b)` and it is the fractional length of the projection along `onto`.

Reply:
The comment is lifted from the wikipedia page (https://en.wikipedia.org/wiki/Vector_projection) which has this image.
![image](https://user-images.githubusercontent.com/15253010/125387727-bf4ae480-e374-11eb-94fd-b8441c7f85f9.png)

And we are actually doing that math, what I called `scalarProjection` is equal to `[(a·b) / (b·b)]` which is then multiplied by `b` in the next lines.

**However** I agree that that is definitely not the scalar projection as the scalar projection is (again from Wikipedia https://en.wikipedia.org/wiki/Scalar_projection#Definition_in_terms_of_a_and_b)
![image](https://user-images.githubusercontent.com/15253010/125387949-09cc6100-e375-11eb-8e2c-70540ab4f45a.png)

Resolving this and accepting your other suggestion to rename the variable as you are right and this would be the `normalizedScalarProjection` or _"the fractional length of the projection along `onto`"_